docs/source/examples/example_doc_generator.py

```diff
@@ -82,69 +115,68 @@ def generate_example_doc(app, docs_path, script_path):
     relative_script_path = script_path[
         script_path.find('examples'):].replace('\\', '/')
 
-    with open(os.path.join(script_path)) as fid:
-        script_text = fid.read()
-
-    temp_path = os.path.join(docs_path, os.path.basename(script_path))
+    # Add the script to the Python Path
+    old_python_path = sys.path
+    sys.path = sys.path + [os.path.dirname(script_path)]
 
-    with open(temp_path, 'wb') as fid:
-        fid.write(script_text.encode())
-
-    snapshot_success = True
+    snapshot_success = False
     mod = None
     with enaml.imports():
         try:
             mod = __import__(script_name)
+            save_snapshot_of_module(mod, image_path)
+            snapshot_success = True
         except Exception as err:
-            print('Could not create: %s' % script_name)
+            print('Could not snapshot: %s' % script_name)
             print('    %s' % err)
-            os.remove(temp_path)
-            snapshot_success = False
-
-    if mod:
-        try:
-            view = mod.Main()
-            snapshot = SnapShot(path=image_path, view=view)
-            view.show()
-            QTest.qWaitForWindowExposed(view.proxy.widget)
-            QTest.qWait(500)
-            snapshot.snapshot()
-            QTest.qWait(100)
-        except Exception as err:
-            print('Could not create: %s' % script_name)
-            print('    %s' % err)
-            snapshot_success = False
         finally:
-            os.remove(temp_path)
+            # The import leaves behind a cache. Clean it up.
+            enaml_cache_dir = os.path.join(
+                os.path.dirname(script_path), '__enamlcache__')
+            shutil.rmtree(enaml_cache_dir)
 
-    docstring = script_text[script_text.find('"""') + 3:]
-    docstring = docstring[: docstring.find('"""')]
-    docstring = docstring.replace('<< autodoc-me >>\n', '').strip()
+    # Restore Python path.
+    sys.path = old_python_path
+
+    with open(os.path.join(script_path)) as fid:
+        script_text = fid.read()
+
+    docstring = clean_docstring(extract_docstring(script_text))
+
+    screenshot_rst = '' if not snapshot_success else ("""
+Screenshot
```

Comment:
Using `textwrap.dedent` or separate global variables for the template strings might make this easier to read.

Reply:
Thank you for the introduction to textwrap.dedent(). I never knew that feature was hidden in that module! Hope to use it every time I am embedding long strings now.

Reply:
@MatthieuDartiailh introduced it to me so thank him :+1: